contrib/externs/angular-1.5.js

```diff
@@ -2468,3 +2467,41 @@ angular.$timeout_.cancel = function(promise) {};
 
 /** @typedef {!Window} */
 angular.$window;
+
+/******************************************************************************
+ * $locale Service
+ *****************************************************************************/
+
+ /**
+ * @typedef {{
+ *   DATETIME_FORMATS: {
```

Comment:
According to the Angular docs, the only public API is the id field:
https://docs.angularjs.org/api/ng/service/$locale

Do you really need all those nested fields? They are supposed to be an implementation detail.

```js
/** @interface */
angular.$locale = function() {};
/** @type {string} */
angular.$locale.prototype.id;
```